util/src/main/java/com/coderanch/util/cli/InputUtility.java

```diff
@@ -0,0 +1,277 @@
+        }
+        catch (NumberFormatException e) {
+            return OptionalDouble.empty();
+        }
+    }
+
+    /**
+     * Gets the next yes or no answer.
+     * Calls {@link InputUtility#nextString(String, Predicate)}.
+     * Passes the prompt and a predicate that validates "y", "n", "yes", "no".
+     *
+     * @param prompt the prompt to display to the user.
+     * @return true for yes and false for no.
+     * @throws IOException when there's a problem with {@link InputStream}
+     */
+    public boolean nextYesNo(String prompt) throws IOException {
+        var result = this.nextString(prompt, yesOrNo()).trim().toLowerCase();
+        return YES_SYNONYMS.contains(result);
+    }
+
+    /**
+     * Calls {@link InputUtility#pause(String)}.
+     * The prompt defaults to "Press when ready".
+     *
+     * @throws IOException when there's a problem with {@link InputStream}
+     */
+    public void pause() throws IOException {
+        pause("Press when ready");
+    }
+
+    /**
+     * Calls nextString{@link InputUtility#nextString(String, Predicate)}.
+     * Uses prompt entered and waits for the user to press <enter>.
+     *
+     * @param prompt the prompt to display to the user.
+     * @throws IOException when there's a problem with {@link InputStream}
+     */
+    public void pause(String prompt) throws IOException {
+        this.nextString(prompt, String::isBlank);
+    }
+
+    /**
+     * Get a String predicate that tests for (case insensitive) "y", "n", "yes", "no".
+     *
+     * @return a String predicate that tests for (case insensitive) "y", "n", "yes", "no".
+     */
+    public static Predicate<String> yesOrNo() {
+        return s -> {
+            var cleanedString = s.trim().toLowerCase();
+            if (YES_SYNONYMS.contains(cleanedString)) {
+                return true;
+            }
+            else {
+                return NO_SYNONYMS.contains(cleanedString);
+            }
+        };
+    }
+
+    /**
+     * Get a String predicate that takes two or more Strings and tests whether any one of them match the input.
+     *
+     * @param these the strings for the input to match to.
+     * @return a String predicate that takes two or more Strings and tests whether any one of them match the input.
+     */
+    public static Predicate<String> oneOfThese(String... these) {
+        if (these.length < 2) {
+            throw new IllegalArgumentException("Must be two or more Strings.");
+        }
+        return s -> Arrays.stream(these)
+            .anyMatch(choice -> choice.equalsIgnoreCase(s.trim()));
+    }
+
+    /**
+     * Get an int predicate that tests whether the input is between the two limits.
+     *
+     * @param lower the lower bound
+     * @param upper the exclusive upper bound
+     * @return an int predicate that tests whether the input is between the two limits.
+     */
+    public static Predicate<Integer> intRange(int lower, int upper) {
+        if (lower > upper) {
```

Comment:
What would happen if you created and used a predicate where the lower bound is equal to the upper bound?

Reply:
it would never be true